frontend/src/app/firewood/forests/buy-firewood-permit/buy-firewood-permit.component.ts

```diff
@@ -0,0 +1,263 @@
+      name: 'description',
+      content: `Purchase a Firewood permit with the United States Forest Service on your National Forest.`
+    });
+    this.applicationForm = new FormGroup({
+      acceptPII: new FormControl(),
+      numberOfCords: new FormControl(),
+      firstName: new FormControl(),
+      lastName: new FormControl()
+    });
+  }
+
+  /**
+   * Update total cost when quantity changes
+   */
+  quantityChange(value) {
+    this.applicationForm.get('numberOfCords').setValidators([
+      Validators.required,
+      lessThanOrEqualValidator(4, 1)
+    ]);
+    if (!this.applicationForm.get('numberOfCords').errors) {
+      this.updateTotalCost();
+    } else {
+      this.applicationForm.get('totalCost').setValue(0);
+    }
+  }
+
+  /**
+   * @returns application form
+   */
+  getApplicationForm(formBuilder, maxCords) {
+    return formBuilder.group({
+      acceptPII: [false, Validators.required],
+      forestId: ['', [Validators.required]],
+      forestAbbr: [''],
+      // orgStructureCode: ['', [Validators.required]],
+      firstName: ['', [Validators.required, alphanumericValidator(), Validators.maxLength(36)]],
+      lastName: ['', [Validators.required, alphanumericValidator(), Validators.maxLength(60)]],
+      emailAddress: ['', [Validators.required, Validators.email, alphanumericValidator(), Validators.pattern('^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,4}$'), Validators.maxLength(255)]],
+      emailAddressConfirmation: [
+        '', [Validators.required, Validators.email, alphanumericValidator(), Validators.pattern(
+          '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,4}$'
+        ), Validators.maxLength(255)]
+      ],
+      numberOfCords: ['', [Validators.required, Validators.min(1), Validators.max(maxCords)]],
+      // quantity: ['', [Validators.required, Validators.min(1), Validators.max(maxCords)]],
+      totalCost: [0, [Validators.required, currencyValidator()]]
+    },
+    {validator: emailConfirmationValidator('emailAddress', 'emailAddressConfirmation')});
+  }
+
+  /**
+   * Get application form and set default values
+   */
+  createForm(forest, formBuilder) {
+    this.applicationForm = this.getApplicationForm(formBuilder, 4);
+    this.applicationForm.get('acceptPII').setValue(false);
+    this.applicationForm.get('forestId').setValue(forest.id);
+    this.applicationForm.get('forestAbbr').setValue(forest.forestAbbr);
+    // this.applicationForm.get('orgStructureCode').setValue(forest.orgStructureCode);
+
+    if (this.permit) {
+      this.rePopulateForm();
+    }
+
+    this.applicationForm.get('numberOfCords').valueChanges.subscribe(value => {
+      this.quantityChange(value);
+    });
+  }
+
+  /**
+   * Redirect to tree guidelines if forest start date is after today
+   */
+  checkSeasonStartDate(forest) {
+    // if (forest && moment(forest.startDate).isAfter(moment())) {
+    //   this.router.navigate(['/christmas-trees/forests/', forest.forestAbbr]);
+    // }
```

Comment:
Looks like when a user tries to access this page when the season is closed we were redirecting them. We might want to do the same thing here.

Reply:
Leaving this as an open item once we fully implement season dates, we will need to flesh it out and test at that time.